google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TimePartitioning.java

```diff
@@ -49,9 +52,92 @@
     DAY
   }
 
-  private TimePartitioning(Type type, Long expirationMs) {
-    this.type = checkNotNull(type);
-    this.expirationMs = expirationMs;
+  /**
+   * A builder for {@code TimePartitioning} objects.
+   */
+  public abstract static class Builder {
```

Comment:
I'm a little confused by two `Builder`s are necessary. Since this class is mostly a "value class" with only some conversion methods, I think we can make life easier by using `@AutoValue`?

Reply:
I've attempted this, but I've never worked with `@AutoValue`. Let me know if this looks right?